app/js/components/manageApps/ManageApps.jsx

```diff
@@ -360,73 +371,88 @@ export default class ManageApps extends React.Component {
-                  return {
-                    appList: [res.data],
-                    downloadUri: res.data['versions'][0].downloadUri,
+          const searchResults = response.data;
+          if (searchResults.length === 0) {
+            this.setState((prevState, props) => {
+              return {
+                appList: []
+              };
+            });
+          } else {
+            searchResults.forEach((result, index) => {
+              axios.get(`https://addons.openmrs.org/api/v1//addon/${result.uid}`)
+                .then(result => {
+                  resultData.push({
+                    appDetails: result.data,
+                    downloadUri: result.data['versions'][0] ? result.data['versions'][0].downloadUri: null,
                     install: true
-                  };
+                  });
+                  if (index === searchResults.length - 1) {
+                    this.setState((prevState, props) => {
+                      return {             
+                        appList: resultData
+                      };
+                    });
+                  }
                 });
-              });
+            });
+          }
+        })
+        .catch(error => {
+          if (error) {
+            this.setState((prevState, props) => {
+              return {
+                appList: []
+              };
+            });
           }
         });
-      }
     } else {
       this.setState((prevState, props) => {
         return {
-          appList: staticAppList,
-          searchResults: []
+          appList: staticAppList
         };
       });
     }
   }
 
-  searchAddOn(event) {
-    event.preventDefault();
-    const searchValue = event.target.value;
-    const { searchResults, staticAppList, appList } = this.state;
+  searchAddOn(searchValue) {
+    const resultData = [];     
+    const { staticAppList, appList } = this.state;
 
     let addOnFound = this.state.staticAppList.filter((app) =>
-      app.name.toLowerCase().indexOf(event.target.value.toLowerCase()) !== -1
-      || app.description.toLowerCase().indexOf(event.target.value.toLowerCase()) !== -1
-      || app.developer["name"].toLowerCase().indexOf(event.target.value.toLowerCase()) !== -1
-      || app.version.indexOf(event.target.value) !== -1);
-
-    this.onlineSearchHandler(addOnFound, searchValue, searchResults, staticAppList);
+      app.appDetails.name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1
+      || app.appDetails.description.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1
+      || app.appDetails.developer["name"].toLowerCase().indexOf(searchValue.toLowerCase()) !== -1
+      || app.appDetails.version.indexOf(searchValue) !== -1);
+    if (addOnFound.length > 0) {
+      resultData.push({
+        appDetails: addOnFound,
+        downloadUri: null,        
+        install: false
+      });
+    }
+    if(resultData,length > 0) {
+      this.setState((prevState, props) => {
+        return {
```

Comment:
Is 'resultData,length' really what you meant to write or 'resultData.length'

Reply:
Wow, thanks a lot. @jacksono would correct this `typo` asap